### PR TITLE
Add tests for List (Pair 2)

### LINKED ISSUE #36: Write JUnit test for List (Pair 2)
According to #17  we in pair 2 should write junit test for the following methods

- [x] `public void __setitem__(org.python.Object index, org.python.Object value)`
- [x] `public void __delitem__(org.python.Object index)`
- [x] `public org.python.Object __iter__()`
- [x] `public org.python.Object __reversed__()`
- [x] `public org.python.Object __contains__(org.python.Object item)`
- [x] `public org.python.Object __add__(org.python.Object other)`
- [x] `public org.python.Object __mul__(org.python.Object other)`
- [x] `public org.python.Object __imul__(org.python.Object other)` SAME IMPLEMENTATION AS MUL?
- [x] ~`public org.python.Object __rmul__(org.python.Object other)`~ NOT IMPLEMENTED
- [x] `public org.python.Object append(org.python.Object item)`
- [x] `public org.python.Object clear()`
- [x] `public org.python.Object copy()`
- [x] `public org.python.Object count(org.python.Object other)`
- [x] `public org.python.Object extend(org.python.Object other)`
- [x] `public org.python.Object index(org.python.Object item, org.python.Object start, org.python.Object end)`
- [x] `public org.python.Object insert(org.python.Object index, org.python.Object item)`
- [x] `public org.python.Object pop(org.python.Object item)`
- [x] `public org.python.Object remove(org.python.Object item)`
- [x] `public org.python.Object reverse()`
- [x] `public org.python.Object sort(final org.python.Object key, org.python.Object reverse)`
- [x] ~`public int compare(org.python.Object o1, org.python.Object o2)`~ NESTED HELPER METHOD FOR SORT()
- [x] ~`public org.python.Object __round__(org.python.Object ndigits)`~ NOT IMPLEMENTED NOR TESTED

### PR DESCRIPTION
**Pull request by rehnarama**
Fixes #36 

To run the tests, simply run `make test`

Don't forget to squash and merge when accepting PR

```
Co-authored-by: Adam Inersjö <adam.inersjo.2778@student.uu.se>
Co-authored-by: Joel Westerlund <jowe3751@student.uu.se>
Co-authored-by: Michael Rehn <rehn.michael@outlook.com>
```

### CODE REVIEW

File: python/common/org/python/types/ListTest.java [hunk 1]
```diff
@@ -1,14 +1,635 @@
...
 
 import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.assertFalse;
 import org.junit.Test;
 
+
 public class ListTest {
 
-    // Simple stub to test if compilation of tests works
```
Comment (JonasNorlinder) [inline python/common/org/python/types/ListTest.java]: Remove incorrect comment.
```suggestion
```

File: python/common/org/python/types/ListTest.java [hunk 2]
```diff
@@ -1,14 +1,635 @@
...
+    public void SetItemInteger() {
+        List l = createList(1);
+
+        l.__setitem__(Int.getInt(0), Int.getInt(5));
+        assertEquals(Int.getInt(5), l.__getitem__(Int.getInt(0)));
     }
-}
 
+    @Test
+    public void SetItemString() {
```
Comment (JonasNorlinder) [inline python/common/org/python/types/ListTest.java]: I prefer our (pair 1) naming convention: `methodName_thingToTest`

**Review by JonasNorlinder — CHANGES_REQUESTED** [review]

**Comment by mawkler** [comment]
Many methods don't test the trivial edge cases like empty lists. I think you should  add that edge case for `__contains__()`, `__add__()`, `copy()`, `count()`, `extend()`, `insert()`, `reverse()` and `sort()`.

**Review by mawkler — APPROVED** [review]
Looks good!
